caravel/assets/javascripts/components/CopyToClipboard.jsx

```diff
@@ -0,0 +1,102 @@
+import React, { PropTypes } from 'react';
+import { Button, Tooltip, OverlayTrigger } from 'react-bootstrap';
+
+const propTypes = {
+  copyNode: PropTypes.node,
+  onCopyEnd: PropTypes.func,
+  shouldShowText: PropTypes.bool,
+  text: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  copyNode: <span>Copy</span>,
+  onCopyEnd: () => {},
+  shouldShowText: true,
+};
+
+export default class CopyToClipboard extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      hasCopied: false,
+    };
+
+    // bindings
+    this.copyToClipboard = this.copyToClipboard.bind(this);
```

Comment:
sucks that we need to write this boilerplate, but haven't the `.bind` in render isn't better...